### PR TITLE
Add Knative Example

### PR DESCRIPTION
**Pull request by hemikak**

### CODE REVIEW

File: kubernetes-extension-examples/examples/knative-deployment/knative_deployment.bal [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import ballerina/http;
+import ballerina/knative;
+import ballerina/log;
+
+//Add the `@knative:Service` to a Ballerina service to generate a Knative Service artifact and push the docker image to docker hub
```
Comment (praneesha) [inline kubernetes-extension-examples/examples/knative-deployment/knative_deployment.bal]:
```suggestion
//Add the `@knative:Service` to a Ballerina service to generate a Knative Service artifact and push the Docker image to Docker Hub.
```

File: kubernetes-extension-examples/examples/knative-deployment/knative_deployment.bal [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+@knative:Service {
+    //Enable pushing the docker image.
```
Comment (praneesha) [inline kubernetes-extension-examples/examples/knative-deployment/knative_deployment.bal]:
```suggestion
    //Enable pushing the Docker image.
```

File: kubernetes-extension-examples/examples/knative-deployment/knative_deployment.bal [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    push: true,
+    //Set the name of the docker image.
```
Comment (praneesha) [inline kubernetes-extension-examples/examples/knative-deployment/knative_deployment.bal]:
```suggestion
    //Set the name of the Docker image.
```

File: kubernetes-extension-examples/examples/knative-deployment/knative_deployment.bal [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    name: "hello-world-knative",
+    //Sets username credential to push the docker image using `DOCKER_USERNAME` environment  variable.
```
Comment (praneesha) [inline kubernetes-extension-examples/examples/knative-deployment/knative_deployment.bal]:
```suggestion
    //Sets the username credential to push the Docker image using the `DOCKER_USERNAME` environment  variable.
```

File: kubernetes-extension-examples/examples/knative-deployment/knative_deployment.bal [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    username: "$env{DOCKER_USERNAME}",
+    //Sets password credential to push the docker image using `DOCKER_PASSWORD` environment  variable.
```
Comment (praneesha) [inline kubernetes-extension-examples/examples/knative-deployment/knative_deployment.bal]:
```suggestion
    //Sets the password credential to push the Docker image using the `DOCKER_PASSWORD` environment  variable.
```

File: kubernetes-extension-examples/examples/knative-deployment/knative_deployment.bal [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    password: "$env{DOCKER_PASSWORD}",
+    //Setting the registry url.
```
Comment (praneesha) [inline kubernetes-extension-examples/examples/knative-deployment/knative_deployment.bal]:
```suggestion
    //Setting the registry URL.
```

File: kubernetes-extension-examples/examples/knative-deployment/knative_deployment.description [hunk 7]
```diff
@@ -0,0 +1,3 @@
+//Ballerina supports generating Knative Serivces artifacts based on annotations.
+//Artifacts willbe generated by adding the  @knative:Service annotation on the HTTP service.
```
Comment (praneesha) [inline kubernetes-extension-examples/examples/knative-deployment/knative_deployment.description]:
```suggestion
//Artifacts will be generated by adding the `@knative:Service` annotation to the HTTP service.
```

File: kubernetes-extension-examples/examples/knative-deployment/knative_deployment.out [hunk 8]
```diff
@@ -0,0 +1,37 @@
+# Build the ballerina program
```
Comment (praneesha) [inline kubernetes-extension-examples/examples/knative-deployment/knative_deployment.out]:
```suggestion
# Build the Ballerina program.
```

File: kubernetes-extension-examples/examples/knative-deployment/knative_deployment.out [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+$ ballerina build knative_deployment.bal
+Compiling source
```
Comment (praneesha) [inline kubernetes-extension-examples/examples/knative-deployment/knative_deployment.out]:
```suggestion
Compiling the source.
```

File: kubernetes-extension-examples/examples/knative-deployment/knative_deployment.out [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+	knative_deployment.bal
+
+Generating executables
```
Comment (praneesha) [inline kubernetes-extension-examples/examples/knative-deployment/knative_deployment.out]:
```suggestion
Generating executables.
```

File: kubernetes-extension-examples/examples/knative-deployment/knative_deployment.out [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+	knative_deployment.jar
+
+Generating Knative artifacts...
+
+	@knative:Service 			 - complete 1/1
+	@knative:Docker 			 - complete 3/3
+
+	Run the following command to deploy the Knative artifacts:
```
Comment (praneesha) [inline kubernetes-extension-examples/examples/knative-deployment/knative_deployment.out]:
```suggestion
	Execute the below command to deploy the Knative artifacts:
```

File: kubernetes-extension-examples/examples/knative-deployment/knative_deployment.out [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+	kubectl apply -f ./kubernetes
+
+	Run the following command to install the application using Helm:
```
Comment (praneesha) [inline kubernetes-extension-examples/examples/knative-deployment/knative_deployment.out]:
```suggestion
	Execute the below command to install the application using Helm:
```

File: kubernetes-extension-examples/examples/knative-deployment/knative_deployment.out [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+	helm install --name helloworld-svc ./kubernetes/helloworld-svc
+
+
+# Verify the Docker image is generated
```
Comment (praneesha) [inline kubernetes-extension-examples/examples/knative-deployment/knative_deployment.out]:
```suggestion
# Verify if the Docker image is generated.
```

File: kubernetes-extension-examples/examples/knative-deployment/knative_deployment.out [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+$ docker images
+REPOSITORY           TAG                 IMAGE ID            CREATED             SIZE
+knative_deployment   latest              85933f1c05b0        3 minutes ago       133MB
+
+# Apply the knative artifacts
```
Comment (praneesha) [inline kubernetes-extension-examples/examples/knative-deployment/knative_deployment.out]:
```suggestion
# Apply the Knative artifacts.
```

File: kubernetes-extension-examples/examples/knative-deployment/knative_deployment.out [hunk 15]
```diff
@@ -0,0 +1,37 @@
...
+$ kubectl apply -f ./kubernetes/
+service.serving.knative.dev/helloworld-svc created
+
+# Wait for the hello-world-knative route to be ready
```
Comment (praneesha) [inline kubernetes-extension-examples/examples/knative-deployment/knative_deployment.out]:
```suggestion
# Wait for the `hello-world-knative` route to be ready.
```

File: kubernetes-extension-examples/examples/knative-deployment/knative_deployment.out [hunk 16]
```diff
@@ -0,0 +1,37 @@
...
+$ watch -n 1 kubectl get pod,ksvc,configuration,revision,route
+NAME                                            URL                                              READY   REASON
+route.serving.knative.dev/hello-world-knative   http://hello-world-knative.default.example.com   True
+
+# Access the service
```
Comment (praneesha) [inline kubernetes-extension-examples/examples/knative-deployment/knative_deployment.out]:
```suggestion
# Access the service.
```

File: kubernetes-extension-examples/examples/kubernetes-deployment/kubernetes_deployment.out [hunk 17]
```diff
@@ -27,15 +27,15 @@ Generating artifacts...
 	kubectl apply -f ./kubernetes
 
 	Run the following command to install the application using Helm:
```
Comment (praneesha) [inline kubernetes-extension-examples/examples/kubernetes-deployment/kubernetes_deployment.out]:
```suggestion
	Execute the below command to install the application using Helm: 
```

File: kubernetes-extension-examples/examples/kubernetes-deployment/kubernetes_deployment.out [hunk 18]
```diff
@@ -27,15 +27,15 @@ Generating artifacts...
...
+	helm install --name kubernetes-deployment-deployment ./kubernetes/kubernetes-deployment-deployment
 
 
 # Verify the Docker image is generated
 $ docker images
 REPOSITORY  TAG      IMAGE ID            CREATED             SIZE
 kubernetes  v1.0   6c0a26a62545        2 seconds ago       127MB
 
-# Run the generated Docker image
+# Apply the kubernetes artifacts
```
Comment (praneesha) [inline kubernetes-extension-examples/examples/kubernetes-deployment/kubernetes_deployment.out]:
```suggestion
# Apply the Kubernetes artifacts.
```

File: kubernetes-extension-examples/examples/openshift-deployment/openshift_deployment.out [hunk 19]
```diff
@@ -1,13 +1,16 @@
 # Build the ballerina program
 $ ballerina build openshift_deployment.bal
 Compiling source
-    openshift_deployment.bal
-Generating executable
-    openshift_deployment.balx
+	openshift_deployment.bal
+
+Generating executables
```
Comment (praneesha) [inline kubernetes-extension-examples/examples/openshift-deployment/openshift_deployment.out]:
```suggestion
Generating executables.
```

File: kubernetes-extension-examples/examples/openshift-deployment/openshift_deployment.out [hunk 20]
```diff
@@ -1,13 +1,16 @@
 # Build the ballerina program
```
Comment (praneesha) [inline kubernetes-extension-examples/examples/openshift-deployment/openshift_deployment.out]:
```suggestion
# Build the Ballerina program.
```

File: kubernetes-extension-examples/examples/openshift-deployment/openshift_deployment.out [hunk 21]
```diff
@@ -1,13 +1,16 @@
...
 $ ballerina build openshift_deployment.bal
 Compiling source
```
Comment (praneesha) [inline kubernetes-extension-examples/examples/openshift-deployment/openshift_deployment.out]:
```suggestion
Compiling the source.
```

File: kubernetes-extension-examples/examples/openshift-deployment/openshift_deployment.out [hunk 22]
```diff
@@ -34,36 +37,69 @@ route.route.openshift.io/helloep-openshift-route created
 # Trigger a build to create the Docker image
```
Comment (praneesha) [inline kubernetes-extension-examples/examples/openshift-deployment/openshift_deployment.out]:
```suggestion
# Trigger a build to create the Docker image.
```

File: kubernetes-extension-examples/examples/openshift-deployment/openshift_deployment.out (hunk 21 above)
Comment (hemikak) [reply kubernetes-extension-examples/examples/openshift-deployment/openshift_deployment.out]: This is coming from the build command.

File: kubernetes-extension-examples/examples/knative-deployment/knative_deployment.out (hunk 9 above)
Comment (hemikak) [reply kubernetes-extension-examples/examples/knative-deployment/knative_deployment.out]: These are outputs of the build command.

File: kubernetes-extension-examples/examples/knative-deployment/knative_deployment.out (hunk 10 above)
Comment (hemikak) [reply kubernetes-extension-examples/examples/knative-deployment/knative_deployment.out]: These are outputs of the build command.

File: kubernetes-extension-examples/examples/knative-deployment/knative_deployment.out (hunk 12 above)
Comment (hemikak) [reply kubernetes-extension-examples/examples/knative-deployment/knative_deployment.out]: These are outputs of the build command with the annotations.

File: kubernetes-extension-examples/examples/knative-deployment/knative_deployment.out (hunk 11 above)
Comment (hemikak) [reply kubernetes-extension-examples/examples/knative-deployment/knative_deployment.out]: These are outputs of the build command with annotations.

File: kubernetes-extension-examples/examples/kubernetes-deployment/kubernetes_deployment.out (hunk 17 above)
Comment (hemikak) [reply kubernetes-extension-examples/examples/kubernetes-deployment/kubernetes_deployment.out]: These are outputs of the build command with annotations.

**Pull request merged**